src/components/studies/StudyBuilder.tsx

```diff
@@ -138,6 +141,58 @@ const StudyBuilder: FunctionComponent<StudyBuilderProps> = ({
   const studyDataUpdateFn = useStudyInfoDataDispatch()
   const [open, setOpen] = React.useState(true)
   const [displayBanner, setDisplayBanner] = React.useState(false)
+  const [bannerTypes, setBannerTypes] = React.useState(
+    new Map<
+      string,
+      {
+        bgColor: string
+        displayText: string[]
+        icon: string[]
+        textColor: string
+      }
+    >()
+  )
+
+  React.useEffect(() => {
+    const bannerMap = new Map()
```

Comment:
Banner map is a constant so it would be better defined outside the components. May be even put it into a separate .ts file and import it?